### PR TITLE
Return file-info object

### PR DESCRIPTION
**Pull request by purplecabbage**
Return file-info object with etag, contentLength, creationTime, contentType, ... from list and getProperties calls

## Description

List will return an array of:
{ name, creationTime, lastModified, etag, contentLength, contentType, isDirectory, isPublic, url }

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: lib/impl/AzureBlobFiles.js [hunk 1]
```diff
@@ -177,6 +177,24 @@ class AzureBlobFiles extends Files {
     return azureProps
   }
 
+  async getFileInfo (filePath) {
```
Comment (moritzraho) [inline lib/impl/AzureBlobFiles.js:180]: I would make this function private and also add it into`lib/Files.js` in the `
  /* **************************** PRIVATE ABSTRACT METHODS TO IMPLEMENT ***************************** */` section. I think it should throw not implemented like other primitives and have some JSDOC to define the  interface (including return object -> typedef)

`getFileInfo` is part of the SPI now, it is a primitive that must be implemented by classes extending Files. (Files and subclasse(s) use the template pattern)

File: lib/impl/AzureBlobFiles.js [hunk 2]
```diff
@@ -191,18 +209,36 @@ class AzureBlobFiles extends Files {
...
         // prefix='' works with SAS credentials but breaks with
         // storageAccount credentials => bug in azure?
-        if (_filePath !== '') options.prefix = _filePath
+        if (_filePath !== '') {
+          options.prefix = _filePath
+        }
         const response = await this._wrapProviderRequest(azureProps.containerURL.listBlobFlatSegment(this._azure.aborter, marker, options), { filePath: _filePath })
         marker = response.marker
-        elements = elements.concat(response.segment.blobItems.map(blob => blob.name))
+        elements = elements.concat(response.segment.blobItems.map(blob => {
```
Comment (moritzraho) [inline lib/impl/AzureBlobFiles.js:217]: Same here the `_listFolder's` JSDoc in `Files.js` should be updated with expected new return format (SPI).
Also `list` JSDoc return value should be updated too (API)

**Review by moritzraho — CHANGES_REQUESTED** [review]
One case is missing for list(file) when file is not a directory https://github.com/adobe/aio-lib-files/blob/d6ab8322a8e1ecdc08788625a1dc7222f2a1ff1c/lib/Files.js#L386 to be consistent in that case we should return this.getFileInfo(file)

**Review by moritzraho — COMMENTED** [review]
thanks for the changes, they LGTM 🙂 
(will approve when unit tests are added)

**Comment by shazron** [comment]
unit tests are failing. GH Actions reports `Test Suites: 2 failed, 1 passed, 3 total
Tests:       12 failed, 197 passed, 209 total`

**Pull request merged**